### PR TITLE
Add support of new files/graphs via Insert Data/Insert Where/Insert Delete Where

### PR DESCRIPTION
**Pull request by splattater**
Add support for creation of new named graphs via INSERT DATA/INSERT WHERE/INSERT DELETE WHERE.
Therefor the graph configuration of QuitStore has to be updated for each new commit and created named graphs must lead do adds of the corresponding RDF file blobs (+ their .graph-file blobs) or the quit.config.ttl-blob.

- [x] refactor Config
  - [x] Split QuitConfiguration in QuitStoreConfiguration (initial QuitStore arguments) and QuitGraphConfiguration (dynamic graph-uri to blob mapping)
  - [x] use git objects instead of file system for config
- [x] update blob handling
  - [x] remove unassigned.nq  
  - [x] update behaviour of commit()-method. Add unknown graph uris (new blob) to QuitStore
  - [x] track graph configration in git repo (quit.config.ttl or .graph-files depending on previous commit)

- [x] Enable test in https://github.com/AKSW/QuitStore/commit/2372ff5aa5c8111f0285b40ae308ee8a6e741bef

- [x] Parts of #81: This won't be solved completely by this pull requests. But we will solve the following parts:
  - [x] split of configuration into store/app configuration and graph configuration (per commit)
  - [x] add new files (nq + graph) or add/update files (nq and config.ttl) if untracked named graphs appear

- [x] Write test to not collide with existing files

### CODE REVIEW

File: quit/conf.py [hunk 1]
```diff
@@ -25,14 +26,16 @@ class Feature:
 
 
 class QuitConfiguration:
-    """A class that keeps track of the relation between named graphs and files."""
```
Comment (white-gecko) [inline quit/conf.py]: Why do we still need this class? What does it do? Please add some documentation string.

File: quit/conf.py [hunk 2]
```diff
@@ -93,75 +87,172 @@ def __initstoreconfig(self, namespace, repository, targetdir, configfile, config
...
+                    <{predicate_prefix}> ?prefix ;
+                    <{predicate_namespace}> ?namespace .
+            }}
+        }}""".format(
+            binding=ns['Binding'], predicate_prefix=ns['prefix'],
+            predicate_namespace=ns['namespace']
+        )
+
+        result = self.sysconf.query(q)
+        return [(row['prefix'], row['namespace']) for row in result]
```
Comment (white-gecko) [inline quit/conf.py]: For which reason do we need these bindings?

File: quit/core.py [hunk 3]
```diff
@@ -428,19 +444,44 @@ def build_message(message, kwargs):
...
+                except KeyError:
+                    pass
+            return blobs_new
+
+        def _applyUnknownGraphs(delta):
+            new_contexts = {}
+            for entry in delta:
+                for identifier, changeset in entry.items():
+                    if isinstance(identifier, BNode) or str(identifier) == 'default':
+                        continue  # TODO
```
Comment (white-gecko) [inline quit/core.py]: Is this the default graph case?

File: quit/conf.py (hunk 1 above)
Comment (splattater) [reply quit/conf.py]: This was a (very) small superclass. I removed it.

File: quit/core.py (hunk 3 above)
Comment (splattater) [reply quit/core.py]: Yes, comment added.

File: quit/core.py [hunk 4]
```diff
@@ -428,19 +444,44 @@ def build_message(message, kwargs):
...
+            return blobs_new
+
+        def _applyUnknownGraphs(delta):
+            new_contexts = {}
+            for entry in delta:
+                for identifier, changeset in entry.items():
+                    if isinstance(identifier, BNode) or str(identifier) == 'default':
+                        continue  # TODO default graph use case
+
+                    fileName = quote_plus(identifier + '.nq')
```
Comment (white-gecko) [inline quit/core.py]: I think the problem is happening here.

**Comment by white-gecko** [comment]
In which commit did you add the test to not collide with existing files?

**Pull request merged**